src/config.rs

```diff
@@ -221,25 +254,66 @@ impl OrchestratorConfig {
+        // Detectors are configured
         if self.detectors.is_empty() {
-            Err(Error::NoDetectorsConfigured)
-        } else {
-            for (detector_id, detector) in &self.detectors {
-                // Chunker is valid
-                let valid_chunker = detector.chunker_id == DEFAULT_MODEL_ID
-                    || self
-                        .chunkers
-                        .as_ref()
-                        .is_some_and(|chunkers| chunkers.contains_key(&detector.chunker_id));
-                if !valid_chunker {
-                    return Err(Error::DetectorChunkerNotFound {
-                        detector_id: detector_id.clone(),
-                        chunker_id: detector.chunker_id.clone(),
-                    });
+            return Err(Error::NoDetectorsConfigured);
+        }
+
+        // Detector configs are valid
+        for (detector_id, detector) in &self.detectors {
+            // Hostname is valid
+            if !is_valid_hostname(&detector.service.hostname) {
+                return Err(Error::InvalidHostname(format!(
+                    "detector `{detector_id}` has an invalid hostname"
+                )));
+            }
+            // Chunker is valid
+            let valid_chunker = detector.chunker_id == DEFAULT_MODEL_ID
+                || self
+                    .chunkers
+                    .as_ref()
+                    .is_some_and(|chunkers| chunkers.contains_key(&detector.chunker_id));
+            if !valid_chunker {
+                return Err(Error::DetectorChunkerNotFound {
+                    detector_id: detector_id.clone(),
+                    chunker_id: detector.chunker_id.clone(),
+                });
+            }
+        }
+
+        // Chunker config is valid
+        if let Some(chunkers) = &self.chunkers {
+            for (chunker_id, chunker) in chunkers {
+                // Hostname is valid
+                if !is_valid_hostname(&chunker.service.hostname) {
+                    return Err(Error::InvalidHostname(format!(
+                        "chunker `{chunker_id}` has an invalid hostname"
+                    )));
                 }
             }
-            Ok(())
         }
+
+        // Generation config is valid
+        if let Some(generation) = &self.generation {
+            // Hostname is valid
+            if !is_valid_hostname(&generation.service.hostname) {
+                return Err(Error::InvalidHostname(
+                    "`generation` has an invalid hostname".into(),
+                ));
+            }
+        }
+
+        // Chat generation config is valid
+        if let Some(chat_generation) = &self.chat_generation {
+            // Hostname is valid
+            if !is_valid_hostname(&chat_generation.service.hostname) {
+                return Err(Error::InvalidHostname(
+                    "`chat_generation` has an invalid hostname".into(),
+                ));
+            }
+        }
```

Comment:
Should we break these into separate functions instead of leaving comments (e.g. `validate_detector_config()`, `validate_chunker_config()` and `validate_generation_config()`?

Reply:
Sure. Just pushed a change to split the rules into methods.

Reply:
Looks great. Thanks!